daml-script/daml3/Daml/Script/Questions/Submit/Error.daml

```diff
@@ -0,0 +1,120 @@
+      userErrorMessage : Text -- TODO[SW] Post LF 1.14, only `submitMustFail` can throw this. I'll refactor that later and likely remove this error
+  | -- | Failure due to false result from `ensure`, strictly pre-exception.
+    TemplatePreconditionViolated -- According to docs, not throwable with LF >= 1.14, unsure what replaces it
+  | -- | Attempted to create a contract with empty contract key maintainers
+    CreateEmptyContractKeyMaintainers with
+      invalidTemplate : AnyTemplate
+  | -- | Attempted to fetch a contract with empty contract key maintainers
+    FetchEmptyContractKeyMaintainers with
+      failedTemplateKey : AnyContractKey
+  | -- | Attempted to exercise/fetch a contract with the wrong template type
+    WronglyTypedContract with
+      contractId : AnyContractId -- ^ Any contract Id of the actual contract
+      expectedTemplateId : TemplateTypeRep
+      actualTemplateId : TemplateTypeRep
+  | -- | Attempted to use a contract as an interface that it does not implement
+    ContractDoesNotImplementInterface with
+      contractId : AnyContractId
+      templateId : TemplateTypeRep
+      interfaceId : TemplateTypeRep -- Consider InterfaceTypeRep from DA.Internal.Interface.AnyView.Types (so exposing it in stdlib)
+  | -- | Attempted to use a contract as a required interface that it does not implement
+    ContractDoesNotImplementRequiringInterface with
+      contractId : AnyContractId
+      templateId : TemplateTypeRep
+      requiredInterfaceId : TemplateTypeRep
+      requiringInterfaceId : TemplateTypeRep
+  | -- | Attempted to compare values that are not comparable
+    NonComparableValues
+  | -- | Illegal Contract ID found in Contract Key
+    ContractIdInContractKey with
+      -- TODO: Update the ContractIdInContractKey scala error to include a TemplateId (which it does have when thrown) in separate PR
+      -- Then we can uncomment this and add the plumbing
+      -- contractKey : AnyContractKey
+  | -- | Attempted to compare a local and global contract ID with the same discriminator. You're doing something very wrong
+    ContractIdComparability with
+      globalExistingContractId : Text
+        -- ^ We do not know the template id at time of comparison. TODO: Consider changing this
+  | -- | Development feature exceptions
+    DevError with -- We just give the error's simple class name and the prettied message it generates. Scripts doesn't need anything more right now.
+      devErrorType : DevErrorType
+      devErrorMessage : Text
+  | -- | Generic catch-all for missing errors.
+    UnknownError with
+      unknownErrorMessage : Text
+  | -- | One of the above error types where the full exception body did not fit into the response, and was incomplete.
+    -- TODO: Should we expose this at all?
+    TruncatedError with
+      truncatedErrorType : Text
+        -- ^ One of the contructor names of SubmitFailure except DevError, UnknownError, TruncatedError
+      truncatedErrorMessage : Text
```

Comment:
This error type exists for when we get errors with missing metadata resources, and can't construct the full error above.
I tried making a contract key with 100,000 of the same party and I didn't get truncated.
I'm not sure how big the error even needs to get to trigger this.
Given its such a pain to even get - should we expose this? Or should we runtime error in the event of truncated errors from the ledger.

Reply:
For anyone skimming: https://github.com/digital-asset/daml/pull/17043/files#r1262245329

Reply:
Yeah I agree